tests/e2e/pageobjects/dashboard/Workspaces.ts

```diff
@@ -68,7 +69,7 @@ export class Workspaces {
 	): Promise<void> {
 		Logger.debug(`"${workspaceName}"`);
 
-		await this.driverHelper.waitVisibility(this.getWorkspaceStatusLocator(workspaceName, WorkspaceStatusUI.Running), timeout);
```

Comment:
Suffix `Locator`is important in function name, because it explains which kind of value the function is actually returning.

Reply:
doesn`t it returns us web element?

>  Typings for lib/by.js Describes a mechanism for locating an element on the page.
>  Locates elements matching a XPath selector. Care should be taken when using an XPath selector with a WebElement as WebDriver will respect the context in the specified in the selector. For example, given the selector //div, WebDriver will search from the document root regardless of whether the locator was used with a WebElement.
> Params:
> xpath – The XPath selector to use.
> Returns:
> !By The new locator.

I can switch it back, no problem

Reply:
but isn`t it sounds right - wait visibility of locator? We had "locator" suffix when this functions were returns us strings. Sorry I am a little bit confused.

Reply:
Workspace status is actually a string with value from the list ["Stopped", "Running", ...].
So, I would expect Workspaces.getWorkspaceStatus() function to return the ready to use actual status of workspace instantly as it without any additional manipulations.

`getWorkspaceStatusLocator()` function in turn is returning locator of status label at the Workspaces page as an object of class "By", which should be applied to another method to get actual status information at the page, e.g. to wait on visibility of `Running` status https://github.com/eclipse/che/blob/main/tests/e2e/pageobjects/dashboard/Workspaces.ts#L71 :
```
await this.driverHelper.waitVisibility(this.getWorkspaceStatusLocator(workspaceName, WorkspaceStatusUI.Running), timeout);
```

So, `getWorkspaceStatusLocator()` != `getWorkspaceStatus()` from this perspective, IMHO.

Reply:
makes sense

Reply:
thank you taking notice into account